procedures.sql

```diff
@@ -0,0 +1,1136 @@
+END;
+GO
+-- EXEC AddCourse 'C#', 10, 50000, 500, 'English', 'Programming'
+
+CREATE PROCEDURE UpdateCourse
+    @course_id INT,
+    @title VARCHAR(50) = NULL,
+    @student_limit INT = NULL,
+    @price INT = NULL,
+    @advance_price INT = NULL,
+    @language VARCHAR(50) = NULL,
+    @subject VARCHAR(100) = NULL
+AS
+BEGIN
+    IF @title IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET title = @title
+            WHERE id = @course_id;
+        END
+    IF @student_limit IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET student_limit = @student_limit
+            WHERE id = @course_id;
+        END
+    IF @price IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET price = @price
+            WHERE id = @course_id;
+        END
+    IF @advance_price IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET advance_price = @advance_price
+            WHERE id = @course_id;
+        END
+    IF @language IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET language = @language
+            WHERE id = @course_id;
+        END
+    IF @subject IS NOT NULL
+        BEGIN
+            UPDATE Course
+            SET subject = @subject
+            WHERE id = @course_id;
+        END
+END;
+GO
+-- EXEC UpdateCourse 1, @title = 'C++', @price = 5000
+
+
+
+CREATE PROCEDURE AddModule
+    @course_id INT,
+    @teacher_id INT,
+    @type VARCHAR(15),
+    @room_id INT = NULL
+AS
+BEGIN
+    INSERT INTO Module (course_id, teacher_id, type, room_id)
+    VALUES (@course_id, @teacher_id, @type, @room_id);
+END;
+GO
+-- EXEC AddModule 12, 23, 'in_person', 17
+
+CREATE PROCEDURE UpdateModule
+    @module_id INT,
+    @course_id INT = NULL,
+    @teacher_id INT = NULL,
+    @type VARCHAR(15) = NULL,
+    @room_id INT = NULL
+AS
+BEGIN
+    IF @course_id IS NOT NULL
+        BEGIN
+            UPDATE Module
+            SET course_id = @course_id
+            WHERE id = @module_id;
+        END
+    IF @teacher_id IS NOT NULL
+        BEGIN
+            UPDATE Module
+            SET teacher_id = @teacher_id
+            WHERE id = @module_id;
+        END
+    IF @type IN ('online_sync', 'online_async')
+        BEGIN
+            UPDATE Module
+            SET type = @type,
+                room_id = NULL
+            WHERE id = @module_id;
+        END
+    IF @type = 'in_person'
+        BEGIN
+            UPDATE Module
+            SET type = @type,
+                room_id = @room_id
+            WHERE id = @module_id;
+        END
+    IF @type = 'hybrid'
+        BEGIN
+            UPDATE Module
+            SET type = @type
+            WHERE id = @module_id;
+        END
+    IF @room_id IS NOT NULL
+        BEGIN
+            UPDATE Module
+            SET room_id = @room_id
+            WHERE id = @module_id;
+        END
+END;
+GO
+-- EXEC UpdateModule 1, @type = 'online_sync'
+
+
+
+CREATE PROCEDURE AddStudy
```

Comment:
https://www.differencebetween.com/difference-between-study-and-vs-studies/

Reply:
thanks